src/systemctl/systemctl-list-units.c

```diff
@@ -641,20 +635,15 @@ static int output_timers_list(struct timer_info *timer_infos, unsigned n) {
                         return table_log_add_error(r);
         }
 
-        if (n > 0) {
-                on = ansi_highlight();
-                off = ansi_normal();
-        } else {
-                on = ansi_highlight_red();
-                off = ansi_normal();
-        }
-
         r = output_table(table);
         if (r < 0)
                 return r;
 
         if (arg_legend != 0) {
-                printf("\n%s%u timers listed.%s\n", on, n, off);
+                printf("\n%s%u timers listed.%s\n",
+                       n > 0 ? ansi_highlight() : ansi_highlight_red(),
+                       n,
+                       ansi_normal());
```

Comment:
I am not sure these are really an improvement, no? Sure, it's a bit shorter, but I kinda liked the fact the coloring was separate from the actual string we output here. Now it's a more complex expression where it's harder to grok what is what...

Reply:
Well, my original patch just reduced scope of the two vars and shortened their definition. The inlining was @yuwata's suggestion. (And the extraction of the legend output to a separate function was an afterthought too.) I'm fine either way.